feature-libs/product-configurator/rulebased/components/attribute/types/read-only/configurator-attribute-read-only.component.ts

```diff
@@ -1,12 +1,54 @@
 import { ChangeDetectionStrategy, Component, Input } from '@angular/core';
 import { Configurator } from '../../../../core/model/configurator.model';
-import { ConfiguratorAttributeBaseComponent } from '../base/configurator-attribute-base.component';
+import { ConfiguratorUiKeyGeneratorService } from '../base/configurator-ui-key-generator.service';
+
 @Component({
   selector: 'cx-configurator-attribute-read-only',
   templateUrl: './configurator-attribute-read-only.component.html',
   changeDetection: ChangeDetectionStrategy.OnPush,
 })
-export class ConfiguratorAttributeReadOnlyComponent extends ConfiguratorAttributeBaseComponent {
+export class ConfiguratorAttributeReadOnlyComponent {
   @Input() attribute: Configurator.Attribute;
+  /**
+   * @deprecated since 4.1: remove redundant input parameter
+   */
   @Input() group: String;
+
+  constructor(
+    protected uiKeyGeneratorService: ConfiguratorUiKeyGeneratorService
+  ) {}
+
+  /**
+   * Creates unique key for config attribute to be sent to configurator.
+   *
+   * @param {Configurator.Attribute} currentAttribute - Attribute
+   * @return {string} - Generated attribute ID
+   */
+  createAttributeIdForConfigurator(
```

Comment:
we can just create a variable, something like this: `attributeId = this.uiKeyGeneratorService.createAttributeIdForConfigurator(this.attribute)`, then in html, using `attributedId` directly.

Reply:
Done! Another method will be deprecated